crates/rover-client/src/shared/check_response.rs

```diff
@@ -1,159 +1,336 @@
 }
 
-impl CheckResponse {
-    pub fn get_json(&self) -> Value {
-        match self {
-            CheckResponse::OperationCheckResponse(operation_check_response) => {
-                operation_check_response.get_json()
-            }
-            CheckResponse::SkipOperationsCheckResponse(operation_less_check_response) => {
-                operation_less_check_response.get_json()
+impl CheckWorkflowResponse {
+    pub fn to_output(&self) -> String {
+        let mut msg = String::new();
+
+        if let Some(core_schema_modified) = self.maybe_core_schema_modified {
+            msg.push('\n');
+            if core_schema_modified {
+                msg.push_str("There were no changes detected in the composed API schema, but the core schema was modified.")
+            } else {
+                msg.push_str("There were no changes detected in the composed schema.")
             }
+            msg.push('\n');
         }
-    }
-}
 
-#[derive(Debug, Serialize, Clone, Eq, PartialEq)]
-pub struct SkipOperationsCheckResponse {
-    pub target_url: Option<String>,
-    pub core_schema_modified: bool,
-}
+        if let Some(operations_response) = &self.maybe_operations_response {
+            if !operations_response.changes.is_empty() {
+                msg.push_str(&Self::task_title(
+                    "Operation Check",
+                    operations_response.task_status.clone(),
+                ));
+                msg.push_str(operations_response.get_output().as_str());
+                msg.push('\n');
+            }
+        }
 
-impl SkipOperationsCheckResponse {
-    pub fn to_output(&self) -> String {
-        let mut msg = if self.core_schema_modified {
-            "There were no changes detected in the composed API schema, but the core schema was modified.".to_string()
-        } else {
-            "There were no changes detected in the composed schema.".to_string()
-        };
+        if let Some(lint_response) = &self.maybe_lint_response {
+            if !lint_response.diagnostics.is_empty() {
+                msg.push_str(&Self::task_title(
+                    "Lint Check",
+                    lint_response.task_status.clone(),
+                ));
+                msg.push_str(lint_response.get_output().as_str());
+                msg.push('\n');
+            }
+        }
 
-        if let Some(url) = &self.target_url {
-            msg.push_str("\n\n");
-            msg.push_str("View full details at: ");
-            msg.push_str(url);
-        };
+        if let Some(downstream_response) = &self.maybe_downstream_response {
+            if !downstream_response.blocking_variants.is_empty() {
+                msg.push_str(&Self::task_title(
+                    "Downstream Check",
+                    downstream_response.task_status.clone(),
+                ));
+                msg.push_str(downstream_response.get_output().as_str());
+                msg.push('\n');
+            }
+        }
```

Comment:
this chunk feels a bit hacky to me - it would make for a more stable solution to just use the `Serialize` trait to make the JSON output directly, like it did before with `json!(self)`